File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProf = 0, buy = prices[0];
    
    for(let sell of prices) {
        if(buy < sell) {
            maxProf = Math.max(maxProf, sell - buy);
        }else {
            buy = sell
        }
    }
    return maxProf;
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.rb
# @param {Integer[]} prices
# @return {Integer}
def max_profit(prices)
    profit, buy = 0, prices[0]
    
    prices.each do |sell|
        if buy < sell
            profit = [profit, sell-buy].max
        else
            buy = sell
        end
    end
    profit
end